crates/sui-framework/packages/sui-framework/sources/tx_context.move

```diff
@@ -124,18 +150,64 @@ public fun get_ids_created(self: &TxContext): u64 {
 
 #[test_only]
 /// Return the most recent created object ID.
-public fun last_created_object_id(self: &TxContext): address {
-    let ids_created = self.ids_created;
-    assert!(ids_created > 0, ENoIDsCreated);
-    derive_id(*&self.tx_hash, ids_created - 1)
+public fun last_created_object_id(_self: &TxContext): address {
+    last_created_id()
 }
+#[test_only]
+native fun last_created_id(): address;
 
 #[test_only]
 public fun increment_epoch_number(self: &mut TxContext) {
-    self.epoch = self.epoch + 1
+    let epoch = self.epoch() + 1;
+    replace(
+        native_sender(),
+        self.tx_hash,
+        epoch,
+        native_epoch_timestamp_ms(),
+        native_ids_created(),
+        native_gas_price(),
+        native_gas_budget(),
+        native_sponsor(),
+    );
 }
 
 #[test_only]
 public fun increment_epoch_timestamp(self: &mut TxContext, delta_ms: u64) {
-    self.epoch_timestamp_ms = self.epoch_timestamp_ms + delta_ms
+    let epoch_timestamp_ms = self.epoch_timestamp_ms() + delta_ms;
+    replace(
+        native_sender(),
+        self.tx_hash,
+        native_epoch(),
+        epoch_timestamp_ms,
+        native_ids_created(),
+        native_gas_price(),
+        native_gas_budget(),
+        native_sponsor(),
+    );
+}
+
+fun option_sponsor(): Option<address> {
+    let sponsor = native_sponsor();
+    if (sponsor.length() == 0) {
+        option::none()
+    } else {
+        option::some(sponsor[0])
+    }
```

Comment:
Have you run the formatter on this file?

Reply:
I do not think so. Should I?

Reply:
no `{`?